vault/plugin_catalog.go

```diff
@@ -432,19 +442,14 @@ func (c *PluginCatalog) UpgradePlugins(ctx context.Context, logger log.Logger) e
 		cmdOld := plugin.Command
 		plugin.Command = filepath.Join(c.directory, plugin.Command)
 
-		pluginType, multiplexingSupport, err := c.getPluginTypeFromUnknown(ctx, logger, plugin)
```

Comment:
Not strictly necessary for the bugfix, but since setInternal always call `getPluginTypeFromUnknown` if `const.PluginTypeUnknown` is provided, this should be safe to remove.